fearless_simd_core/src/lib.rs

```diff
@@ -0,0 +1,339 @@
+//! These are grouped by architecture:
+//!
+//! - [`x86`] contains the tokens for both the x86 and x86-64 targets.
+//!   It also contains tokens for each x86-64 microarchitecture level, see [`x86::V1`] for details.
+//! <!-- TODO: Other architectures -->
+//!
+//! # Examples
+//!
+//! At the time of writing, it is not possible to turn scalar values into <abbr title="Single InstructionMultiple Data">SIMD</abbr>
+//! vector types safely using only the standard library.
+//! These examples use [bytemuck](https://crates.io/crates/bytemuck) for this.
+//!
+//! <!-- TODO -->
+//!
+//! Note that for `aarch64`'s neon, you will want to enable bytemuck's `aarch64_simd` feature.
+//! This is also the case for WASM with `wasm_simd`, but note that this crate
+//! [isn't needed on WASM][attributes.codegen.target_feature.wasm], as it is safe to
+//! call `#[target_features]` on that platform.
+//!
+//! # Crate Feature Flags
+//!
+//! <!-- TODO -->
+//!
+//! # Implementation
+//!
+//! The tokens provided to [`trampoline!`] implement the [`TargetFeatureToken`] trait,
+//! which indicates that a value of that token is only possible to construct if the set
+//! of target features it specifies are enabled.
+//! This means that the macro can use the existence of these token values as
+//! safety proofs that calling a function with those target features is safe.
+//!
+//! This safety proof happens entirely in const evaluation, so if there's a mistake with the
+//! proof, it will cause a compilation error.
+//! The code generated by this macro is thus a function containing the provided code, marked
+//! with `#[target_feature]`, and a call to this newly generated function.
+//!
+//! [attributes.codegen.target_feature.wasm]: https://doc.rust-lang.org/reference/attributes/codegen.html#r-attributes.codegen.target_feature.wasm
+
+// LINEBENDER LINT SET - lib.rs - v4
+// See https://linebender.org/wiki/canonical-lints/
+// These lints shouldn't apply to examples or tests.
+#![cfg_attr(not(test), warn(unused_crate_dependencies))]
+// These lints shouldn't apply to examples.
+#![warn(clippy::print_stdout, clippy::print_stderr)]
+// Targeting e.g. 32-bit means structs containing usize can give false positives for 64-bit.
+#![cfg_attr(target_pointer_width = "64", warn(clippy::trivially_copy_pass_by_ref))]
+// END LINEBENDER LINT SET
+#![cfg_attr(docsrs, feature(doc_cfg))]
+#![no_std]
+
+// TODO: Do we want both an `x86` and `x86_64` module?
+#[cfg(any(target_arch = "x86", target_arch = "x86_64", doc))]
+pub mod x86;
+
+pub mod trampoline;
+
+#[cfg(feature = "std")]
+extern crate std;
+
+/// Token that a set of target feature is available.
+///
+/// Note that this trait is only meaningful when there are values of this type.
+/// That is, to enable the target features in `FEATURES`, you *must* have a value
+/// of this type.
+///
+/// Values which implement this trait are used in the second argument to [`trampoline!`],
```

Comment:
```suggestion
/// Values which implement this trait are used in the first argument to [`trampoline!`],
```

I thought it was passed to the first argument of `trampoline!`